doc/howto/instances_routed_nic_vm.md

```diff
@@ -0,0 +1,42 @@
+(instances-routed-nic-vm)=
+# How to add a routed NIC device to a virtual machine
+
+When adding a [routed NIC device](../reference/devices_nic.md#nictype-routed) to an instance, the instance must be configured to use the link-local gateway IPs as default routes.
+For containers, this is configured for you automatically.
+For virtual machines, the gateways must be configured manually or via a mechanism like `cloud-init`.
+
+To configure the gateways with `cloud-init`, firstly initialize an instance:
+
+    lxc init ubuntu:jammy jammy --vm
+
+Then add the routed NIC device:
+
+    lxc config device add jammy eth0 nic nictype=routed parent=my-parent-network ipv4.address=192.0.2.2 ipv6.address=2001:db8::2
+
+Where `my-parent-network` is your parent network, and the IPv4 and IPv6 addresses are within the subnet of the parent.
+
+Then set the following configuration on the instance:
+
+    lxc config set jammy cloud-init.network-config "$(cat <<EOF
+    network:
+      version: 2
+      ethernets:
+        enp5s0:
+          link-local:
+          - ipv4
+          - ipv6
+          routes:
+          - to: default
+            via: 169.254.0.1
+          - to: default
+            via: fe80::1
+          addresses:
```

Comment:
I don't know if LXD will have a DNS listener on `169.254.0.1`, but if it does, it'd be nice to add a `nameserver` section to the netplan config. If no such listener is created, we should maybe hint the reader that more steps are needed to have DNS resolution working.

Reply:
I'll add a `nameservers` section and tell the reader they need to change the value. I don't know what IP address I should put as the nameserver though.

Reply:
I've added `8.8.8.8` as a nameserver and said that this needs to be changed. I'm not particularly comfortable with that though, because it will not work out of the box.

@ru-fu shall I instead add a note where I am describing the netplan config and say that nameserver configuration is left to the user?

Reply:
Yes just mention that it should be set to a valid DNS server IP.
If there is a lxdbr0 network on the host it can be set to that IP instead.

Reply:
Your suggestion looks good to me. :-)